### PR TITLE
sync pd and tikv configmap in controller

### LINKED ISSUE #1263: sync pd configmap and start script in controller
#1121

### LINKED ISSUE #1264: sync tikv configmap and start script in controller
#1121

Sync the tikv configmap and start-script in controller, an example about syncing these could be found here:

Note:

- Rolling-update of configmap should be controlled via a strategy field.
- An integration test case (e2e) should be added to verify this change will not rolling-update the tikv statefulset created before if there's no change in `tidbcluster` spec.
- Unlike pump, configuration of TiKV must have schema, because TiKV is written in rust, it's hard to import the schema of TiKV's configuration directly, maybe maintain a go copy is an acceptable idea, for example, hackathon team `plainboring` did so : https://github.com/plainboring/config_client/blob/master/pkg/tikv/config.go

### PR DESCRIPTION
**Pull request by aylei**
Signed-off-by: Aylei <rayingecho@gmail.com>

### What problem does this PR solve? 
close #1263 
close #1264 

### Check List 

Tests 

 - Unit test
 - E2E test

Code changes

 - Has Go code change

### Does this PR introduce a user-facing change?:
 
 ```release-note
pd and tikv configs are now managed in TidbCluster resource.
 ```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/pd_config.go [hunk 1]
```diff
@@ -250,7 +246,7 @@ type PDScheduleConfig struct {
 	// MaxStoreDownTime is the max duration after which
 	// a store will be considered to be down if it hasn't reported heartbeats.
 	// +optional
-	MaxStoreDownTime Duration `toml:"max-store-down-time,omitempty" json:"max-store-down-time,omitempty"`
+	MaxStoreDownTime string `toml:"max-store-down-time,omitempty" json:"max-store-down-time,omitempty"`
```
Comment (aylei) [inline pkg/apis/pingcap/v1alpha1/pd_config.go]: `BurntSushi/toml` do not respect `MarshalText`, use `string` instead

Comment (cofyc) [reply pkg/apis/pingcap/v1alpha1/pd_config.go]: `BurntSushi/toml` seems not maintained anymore...

we can use `https://github.com/pelletier/go-toml` and implement `toml.MarshalTOML()` (like MarshalJSON())

Comment (cofyc) [reply pkg/apis/pingcap/v1alpha1/pd_config.go]: https://play.golang.org/p/_SfokqsaCPo it works

Comment (aylei) [reply pkg/apis/pingcap/v1alpha1/pd_config.go]: Yes, that's exactly what I've thought.

Actually I've tried `go-toml`( and `naoina/toml`), it turns out that `go-toml` do not support `interface{}`. Also, PD and TiDB uses `BurntSushi/toml` so there is little verifying work, seems like that `BurntSushi/toml` is not an ideal but the only choice

**Pull request merged**